### PR TITLE
Add region tags to the Go protobuf examples.

### PR DESCRIPTION
**Pull request by tswast**
This will allow us to like to specific snippets of code in the
documentation.  I plan to create a tutorial similar to the C# tutorial
https://developers.google.com/protocol-buffers/docs/csharptutorial

Since that tutorial has sections for populating a proto, parsing, and
serializing, I made a region for each of these for Go.  To make the
populating sample more self-contained, I refactor the listing example
slightly.

### CODE REVIEW

**Comment by tswast** [comment]
@dsymonds Mind taking a look for the Go example changes?

File: examples/list_people_test.go [hunk 1]
```diff
@@ -8,6 +8,50 @@ import (
...
+	want := strings.Split(`Person ID: 1234
+  Name: John Doe
+  E-mail address: jdoe@example.com
+  Home phone #: 555-4321
+`, "\n")
+
+	got := strings.Split(buf.String(), "\n")
+	if len(got) != len(want) {
+		t.Errorf(
+			"writePerson(%s) =>\n\t%q has %d lines, want %d",
```
Comment (dsymonds) [inline examples/list_people_test.go]: this and the next t.Errorf are oddly wrapped. It doesn't help that you've got redundant `.String()` calls. I'd write this on one line:

```go
t.Errorf("writePerson(%s) =>\n\t%q has %d lines, want %d", p, buf, len(got), len(want))
```

But I don't even understand why you split by lines? Why not compare the whole string in one go?

Comment (tswast) [inline examples/list_people_test.go]: True. I was splitting by lines to make it easier to find if there was a mistake somewhere, but this is a small enough string that it isn't necessary. I've made the test much simpler by just comparing the whole string.

**Comment by tswast** [comment]
@dsymonds I updated list_people_test.go. Please take a look.

**Comment by dsymonds** [comment]
Oh, didn't realise you were waiting on me. It was fine apart from those small things I mentioned before.

**Comment by tswast** [comment]
Thanks!

@xfxyjwf Could you take a look / merge?

**Pull request merged**